src/plugins/data/server/search/opensearch_search/opensearch_search_strategy.ts

```diff
@@ -92,7 +105,7 @@ export const opensearchSearchStrategyProvider = (
       } catch (e) {
         if (usage) usage.trackError();
 
-        if (dataSource && request.dataSourceId) {
+        if (dataSource) {
```

Comment:
should the check on `dataSource?.dataSourceEnabled()`

Reply:
That's a good point. Will update. Thanks

Reply:
Updated, thanks.
Resolving this comment